core/modules/taxonomy/tests/taxonomy.test

```diff
@@ -59,14 +59,25 @@ class TaxonomyVocabularyFunctionalTest extends TaxonomyWebTestCase {
     // Visit the main taxonomy administration page.
     $this->backdropGet('admin/structure/taxonomy');
 
+    // Create a new vocabulary and then add terms.
+    $this->clickLink(t('Add vocabulary'));
+    $edit = array();
+    $machine_name = backdrop_strtolower($this->randomName());
+    $edit['name'] = $this->randomName();
+    $edit['description'] = $this->randomName();
+    $edit['machine_name'] = $machine_name;
+    $this->backdropPost(NULL, $edit, t('Save and add terms'));
+    $this->assertRaw(t('Created new vocabulary %name.', array('%name' => $edit['name'])));
+
     // Create a new vocabulary.
+    $this->backdropGet('admin/structure/taxonomy');
     $this->clickLink(t('Add vocabulary'));
     $edit = array();
     $machine_name = backdrop_strtolower($this->randomName());
     $edit['name'] = $this->randomName();
     $edit['description'] = $this->randomName();
     $edit['machine_name'] = $machine_name;
-    $this->backdropPost(NULL, $edit, t('Save'));
+    $this->backdropPost(NULL, $edit, t('Save vocabulary'));
```

Comment:
Similarly to the previous section, should we also be checking that the redirect to the vocabulary listing works here?

Reply:
yes, good idea.